packages/dds/tree/src/test/tablePerformanceTestUtilities.ts

```diff
@@ -0,0 +1,190 @@
+	/**
+	 * The undo stack for the table tree.
+	 */
+	undoStack: Revertible[];
+	/**
+	 * The redo stack for the table tree.
+	 */
+	redoStack: Revertible[];
+	/**
+	 * Unsubscribe from the table tree events and dispose of the undo/redo stacks.
+	 */
+	unsubscribe: () => void;
+}
+
+/**
+ * Factory for creating a table tree schema.
+ * This factory is used to create the schema for the table tree, including cells, columns,
+ * rows, and the table itself.
+ */
+const schemaFactory = new SchemaFactoryAlpha("test");
+
+/**
+ * Defines the schema for a table cell.
+ */
+export class Cell extends schemaFactory.object("table-cell", {
+	cellValue: schemaFactory.string,
+}) {}
+
+/**
+ * Defines the schema for a table column.
+ */
+export class Column extends TableSchema.column({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table row.
+ */
+export class Row extends TableSchema.row({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+/**
+ * Defines the schema for a table, which includes columns and rows.
+ * It uses the previously defined Cell, Column, and Row schemas.
+ */
+export class Table extends TableSchema.table({
+	schemaFactory,
+	cell: Cell,
+	column: Column,
+	row: Row,
+}) {}
+
+/**
+ * Provides a simple table tree initialized with the specified size and cell value.
+ * This helper function creates a table schema, initializes a SharedTree instance,
+ * and populates it with the specified number of rows and columns.
+ * Each cell is initialized with the provided cell value.
+ *
+ * @param tableSize - The number of rows and columns to create in the table.
+ * @param cellValue - The initial value to set in each cell of the table.
+ * @returns A fully initialized table tree definition, including table instance, undo/redo stacks, and a cleanup function.
+ */
+export function createTableTree(tableSize: number, cellValue: string): TableTreeDefinition {
+	const sharedTreeFactory = DefaultTestSharedTreeKind.getFactory();
+	const runtime = new MockFluidDataStoreRuntime({
+		idCompressor: createIdCompressor(),
+		attachState: AttachState.Detached,
+	});
+	const tree = sharedTreeFactory.create(runtime, "tree");
+	const treeView = tree.viewWith(
+		new TreeViewConfiguration({
+			schema: Table,
+			enableSchemaValidation: true,
+		}),
+	);
+
+	treeView.initialize(Table.empty());
+	const undoStack: Revertible[] = [];
+	const redoStack: Revertible[] = [];
+
+	function onDispose(disposed: Revertible): void {
+		const redoIndex = redoStack.indexOf(disposed);
+		if (redoIndex !== -1) {
+			redoStack.splice(redoIndex, 1);
+		} else {
+			const undoIndex = undoStack.indexOf(disposed);
+			if (undoIndex !== -1) {
+				undoStack.splice(undoIndex, 1);
+			}
+		}
+	}
+
+	const unsubscribeFromCommitAppliedEvent = treeView.events.on(
+		"commitApplied",
+		(commit, getRevertible) => {
+			if (getRevertible !== undefined) {
+				const revertible = getRevertible(onDispose);
+				if (commit.kind === CommitKind.Undo) {
```

Comment:
This logic isn't quite right. One issue of note: if a non-undo commit is made, then we should clear the redo stack (otherwise we keep stale redo operations). See this code in our brainstorm example for a more standard way to set this up: https://github.com/microsoft/FluidExamples/blob/main/brainstorm/src/utils/undo.ts#L22

Reply:
I’ve updated the logic to address the issue with stale redo operations. Now, whenever a non-undo commit is made (e.g., a default commit), the redo stack is cleared to ensure no stale operations remain. This aligns with the approach demonstrated in the brainstorm example, where the redo stack is explicitly disposed of and reset during non-undo commits.